pallets/loans/src/loan.rs

```diff
@@ -538,6 +538,60 @@ impl<T: Config> Pallet<T> {
 
         Ok(())
     }
+
+    pub fn reduce_reserves_internal(
+        admin: T::AccountId,
+        currency_id: CurrencyId,
+        reduce_amount: Balance,
+    ) -> DispatchResult {
+        let total_reserves = Self::total_reserves(currency_id);
+        if reduce_amount > total_reserves {
+            return Err(Error::<T>::ReduceReservesFailed.into());
+        }
+
+        let total_reserves_new = total_reserves - reduce_amount;
+        if total_reserves_new > total_reserves {
```

Comment:
I think this will never happen so we can remove this if

Reply:
Hmm, yes. Let me remove it.